krnlmon_main.c

```diff
@@ -2,26 +2,36 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include <stdio.h>
+#if !defined(_GNU_SOURCE)
+#define __USE_GNU
+#endif
 
+#include <pthread.h>
 #include "krnlmon_main.h"
```

Comment:
One generally includes one's own header file before any others. It's a quick-and-dirty check for missing dependencies in the header file.
```
#include "krnlmon_main.h"

#include <pthread.h>
#include <stdio.h>
```

Reply:
Done